### PR TITLE
feat(gov): proposal gating

### PR DESCRIPTION
**Pull request by RexMilce**

### CODE REVIEW

**Comment by Yaroms** [comment]
does not compile and lint @RexMilce

File: x/spec/ante/ante.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+		}
+		// if it's expedited we validate the list of proposal
+		// messages.
+		propMsgs, err := m.GetMsgs()
+		if err != nil {
+			return err
+		}
+		// ensure all proposal exec messages are allowed
+		for i, propMsg := range propMsgs {
+			err = e.ensureInWhitelist(ctx, propMsg)
```
Comment (omerlavanet) [inline x/spec/ante/ante.go:57]: allowlist

**Review by omerlavanet — DISMISSED** [review]

File: app/upgrades/upgrade_0_33_1.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+func v_33_1(
+	m *module.Manager,
+	c module.Configurator,
+	_ BaseAppParamManager,
+	lk *keepers.LavaKeepers,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		// reset whitelist to empty
+		lk.SpecKeeper.WhitelistReset(ctx)
```
[the remark this answers is not in the document]
Comment (Yaroms) [reply app/upgrades/upgrade_0_33_1.go]: https://github.com/lavanet/lava/pull/1109/commits/31c95da5f7c2f6d1b58c1028af5d085396d86ff0

File: x/spec/ante/ante.go (hunk 1 above)
Comment (Yaroms) [reply x/spec/ante/ante.go]: https://github.com/lavanet/lava/pull/1109/commits/31c95da5f7c2f6d1b58c1028af5d085396d86ff0

**Review by Yaroms — APPROVED** [review]

**Pull request merged**